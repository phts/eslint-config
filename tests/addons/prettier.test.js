'use strict'

const helpers = require('../helpers')

describe('addons/prettier', () => {
  helpers.itGeneratesCorrectConfig({
    files: [
      'target.js',
      'target.jsx',
      'target.ts',
      'target.tsx',
      'dir/target.js',
      'dir/target.jsx',
      'dir/target.ts',
      'dir/target.tsx',
    ],
    for: '*.{js,jsx,ts,tsx}',
    extends: ['index', 'addons/prettier'],
  })
})
