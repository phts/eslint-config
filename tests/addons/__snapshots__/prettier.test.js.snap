// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addons/prettier generates correct config for *.{js,jsx,ts,tsx} files 1`] = `
Object {
  "env": Object {},
  "globals": Object {},
  "ignorePatterns": Array [
    "playground/",
  ],
  "noInlineConfig": undefined,
  "parser": null,
  "parserOptions": Object {},
  "plugins": Array [
    "eslint-comments",
    "import",
    "prettier",
  ],
  "reportUnusedDisableDirectives": undefined,
  "rules": Object {
    "@typescript-eslint/indent": Array [
      "off",
    ],
    "@typescript-eslint/member-delimiter-style": Array [
      "off",
    ],
    "accessor-pairs": Array [
      "error",
    ],
    "array-bracket-newline": Array [
      "off",
    ],
    "array-bracket-spacing": Array [
      "off",
    ],
    "array-element-newline": Array [
      "off",
    ],
    "arrow-body-style": Array [
      0,
    ],
    "arrow-parens": Array [
      "off",
    ],
    "arrow-spacing": Array [
      "off",
    ],
    "block-spacing": Array [
      "off",
    ],
    "brace-style": Array [
      "off",
    ],
    "camelcase": Array [
      "error",
    ],
    "comma-dangle": Array [
      "off",
      "always-multiline",
    ],
    "comma-spacing": Array [
      "off",
    ],
    "comma-style": Array [
      "off",
    ],
    "computed-property-spacing": Array [
      "off",
    ],
    "constructor-super": Array [
      "error",
    ],
    "curly": Array [
      0,
    ],
    "dot-location": Array [
      "off",
      "property",
    ],
    "eol-last": Array [
      "off",
    ],
    "eqeqeq": Array [
      "error",
    ],
    "eslint-comments/disable-enable-pair": Array [
      "error",
      Object {
        "allowWholeFile": true,
      },
    ],
    "eslint-comments/no-aggregating-enable": Array [
      "error",
    ],
    "eslint-comments/no-duplicate-disable": Array [
      "error",
    ],
    "eslint-comments/no-unlimited-disable": Array [
      "error",
    ],
    "eslint-comments/no-unused-disable": Array [
      "error",
    ],
    "eslint-comments/no-unused-enable": Array [
      "error",
    ],
    "for-direction": Array [
      "error",
    ],
    "func-call-spacing": Array [
      "off",
    ],
    "function-call-argument-newline": Array [
      "off",
    ],
    "function-paren-newline": Array [
      "off",
    ],
    "generator-star": Array [
      "off",
    ],
    "generator-star-spacing": Array [
      "off",
    ],
    "getter-return": Array [
      "error",
    ],
    "implicit-arrow-linebreak": Array [
      "off",
    ],
    "import/export": Array [
      "error",
    ],
    "import/first": Array [
      "error",
    ],
    "import/named": Array [
      "error",
    ],
    "import/newline-after-import": Array [
      "error",
    ],
    "import/no-absolute-path": Array [
      "error",
    ],
    "import/no-cycle": Array [
      "error",
    ],
    "import/no-named-as-default": Array [
      "error",
    ],
    "import/no-named-as-default-member": Array [
      "error",
    ],
    "import/no-self-import": Array [
      "error",
    ],
    "import/no-useless-path-segments": Array [
      "error",
    ],
    "import/order": Array [
      "error",
    ],
    "indent": Array [
      "off",
      2,
    ],
    "indent-legacy": Array [
      "off",
    ],
    "jsx-quotes": Array [
      "off",
      "prefer-double",
    ],
    "key-spacing": Array [
      "off",
      Object {
        "afterColon": true,
        "beforeColon": false,
        "mode": "minimum",
      },
    ],
    "keyword-spacing": Array [
      "off",
    ],
    "linebreak-style": Array [
      "off",
    ],
    "lines-around-comment": Array [
      0,
    ],
    "max-len": Array [
      0,
      Object {
        "code": 100,
      },
    ],
    "multiline-ternary": Array [
      "off",
    ],
    "new-cap": Array [
      "error",
    ],
    "new-parens": Array [
      "off",
    ],
    "newline-per-chained-call": Array [
      "off",
    ],
    "no-array-constructor": Array [
      "error",
    ],
    "no-arrow-condition": Array [
      "off",
    ],
    "no-async-promise-executor": Array [
      "error",
    ],
    "no-caller": Array [
      "error",
    ],
    "no-case-declarations": Array [
      "error",
    ],
    "no-class-assign": Array [
      "error",
    ],
    "no-comma-dangle": Array [
      "off",
    ],
    "no-compare-neg-zero": Array [
      "error",
    ],
    "no-cond-assign": Array [
      "error",
    ],
    "no-confusing-arrow": Array [
      0,
    ],
    "no-console": Array [
      "error",
      Object {
        "allow": Array [
          "info",
          "warn",
          "error",
        ],
      },
    ],
    "no-const-assign": Array [
      "error",
    ],
    "no-constant-condition": Array [
      "error",
    ],
    "no-control-regex": Array [
      "error",
    ],
    "no-debugger": Array [
      "error",
    ],
    "no-delete-var": Array [
      "error",
    ],
    "no-dupe-args": Array [
      "error",
    ],
    "no-dupe-class-members": Array [
      "error",
    ],
    "no-dupe-else-if": Array [
      "error",
    ],
    "no-dupe-keys": Array [
      "error",
    ],
    "no-duplicate-case": Array [
      "error",
    ],
    "no-duplicate-imports": Array [
      "error",
    ],
    "no-empty": Array [
      "error",
    ],
    "no-empty-character-class": Array [
      "error",
    ],
    "no-empty-pattern": Array [
      "error",
    ],
    "no-eval": Array [
      "error",
    ],
    "no-ex-assign": Array [
      "error",
    ],
    "no-extra-bind": Array [
      "error",
    ],
    "no-extra-boolean-cast": Array [
      "error",
    ],
    "no-extra-parens": Array [
      "off",
      "all",
      Object {
        "ignoreJSX": "all",
      },
    ],
    "no-extra-semi": Array [
      "off",
    ],
    "no-fallthrough": Array [
      "error",
    ],
    "no-floating-decimal": Array [
      "off",
    ],
    "no-func-assign": Array [
      "error",
    ],
    "no-global-assign": Array [
      "error",
    ],
    "no-implied-eval": Array [
      "error",
    ],
    "no-import-assign": Array [
      "error",
    ],
    "no-inner-declarations": Array [
      "error",
    ],
    "no-invalid-regexp": Array [
      "error",
    ],
    "no-irregular-whitespace": Array [
      "error",
    ],
    "no-label-var": Array [
      "error",
    ],
    "no-labels": Array [
      "error",
    ],
    "no-lone-blocks": Array [
      "error",
    ],
    "no-misleading-character-class": Array [
      "error",
    ],
    "no-mixed-operators": Array [
      0,
    ],
    "no-mixed-spaces-and-tabs": Array [
      "off",
    ],
    "no-multi-spaces": Array [
      "off",
    ],
    "no-multiple-empty-lines": Array [
      "off",
    ],
    "no-new-object": Array [
      "error",
    ],
    "no-new-require": Array [
      "error",
    ],
    "no-new-symbol": Array [
      "error",
    ],
    "no-obj-calls": Array [
      "error",
    ],
    "no-octal": Array [
      "error",
    ],
    "no-path-concat": Array [
      "error",
    ],
    "no-proto": Array [
      "error",
    ],
    "no-prototype-builtins": Array [
      "error",
    ],
    "no-redeclare": Array [
      "error",
    ],
    "no-regex-spaces": Array [
      "error",
    ],
    "no-reserved-keys": Array [
      "off",
    ],
    "no-return-assign": Array [
      "error",
    ],
    "no-return-await": Array [
      "error",
    ],
    "no-self-assign": Array [
      "error",
    ],
    "no-self-compare": Array [
      "error",
    ],
    "no-sequences": Array [
      "error",
    ],
    "no-setter-return": Array [
      "error",
    ],
    "no-shadow-restricted-names": Array [
      "error",
    ],
    "no-space-before-semi": Array [
      "off",
    ],
    "no-spaced-func": Array [
      "off",
    ],
    "no-sparse-arrays": Array [
      "error",
    ],
    "no-tabs": Array [
      0,
    ],
    "no-this-before-super": Array [
      "error",
    ],
    "no-throw-literal": Array [
      "error",
    ],
    "no-trailing-spaces": Array [
      "off",
    ],
    "no-undef": Array [
      "error",
    ],
    "no-undef-init": Array [
      "error",
    ],
    "no-unexpected-multiline": Array [
      0,
    ],
    "no-unmodified-loop-condition": Array [
      "error",
    ],
    "no-unneeded-ternary": Array [
      "error",
    ],
    "no-unreachable": Array [
      "error",
    ],
    "no-unsafe-finally": Array [
      "error",
    ],
    "no-unsafe-negation": Array [
      "error",
    ],
    "no-unused-labels": Array [
      "error",
    ],
    "no-unused-vars": Array [
      "error",
    ],
    "no-use-before-define": Array [
      "error",
    ],
    "no-useless-call": Array [
      "error",
    ],
    "no-useless-catch": Array [
      "error",
    ],
    "no-useless-computed-key": Array [
      "error",
    ],
    "no-useless-constructor": Array [
      "error",
    ],
    "no-useless-escape": Array [
      "error",
    ],
    "no-useless-rename": Array [
      "error",
    ],
    "no-var": Array [
      "error",
    ],
    "no-whitespace-before-property": Array [
      "off",
    ],
    "no-with": Array [
      "error",
    ],
    "no-wrap-func": Array [
      "off",
    ],
    "nonblock-statement-body-position": Array [
      "off",
    ],
    "object-curly-newline": Array [
      "off",
    ],
    "object-curly-spacing": Array [
      "off",
      "never",
    ],
    "object-property-newline": Array [
      "off",
      Object {
        "allowAllPropertiesOnSameLine": true,
        "allowMultiplePropertiesPerLine": false,
      },
    ],
    "one-var": Array [
      "error",
      "never",
    ],
    "one-var-declaration-per-line": Array [
      "off",
    ],
    "operator-linebreak": Array [
      "off",
    ],
    "padded-blocks": Array [
      "off",
      "never",
    ],
    "prefer-arrow-callback": Array [
      0,
    ],
    "prefer-const": Array [
      "error",
    ],
    "prettier/prettier": Array [
      "error",
    ],
    "quote-props": Array [
      "off",
      "as-needed",
    ],
    "quotes": Array [
      0,
      "single",
      Object {
        "allowTemplateLiterals": false,
        "avoidEscape": true,
      },
    ],
    "require-yield": Array [
      "error",
    ],
    "rest-spread-spacing": Array [
      "off",
    ],
    "semi": Array [
      "off",
      "never",
    ],
    "semi-spacing": Array [
      "off",
    ],
    "semi-style": Array [
      "off",
    ],
    "space-after-function-name": Array [
      "off",
    ],
    "space-after-keywords": Array [
      "off",
    ],
    "space-before-blocks": Array [
      "off",
    ],
    "space-before-function-paren": Array [
      "off",
      Object {
        "anonymous": "always",
        "asyncArrow": "always",
        "named": "never",
      },
    ],
    "space-before-function-parentheses": Array [
      "off",
    ],
    "space-before-keywords": Array [
      "off",
    ],
    "space-in-brackets": Array [
      "off",
    ],
    "space-in-parens": Array [
      "off",
    ],
    "space-infix-ops": Array [
      "off",
    ],
    "space-return-throw-case": Array [
      "off",
    ],
    "space-unary-ops": Array [
      "off",
    ],
    "space-unary-word-ops": Array [
      "off",
    ],
    "spaced-comment": Array [
      "error",
    ],
    "strict": Array [
      "error",
    ],
    "switch-colon-spacing": Array [
      "off",
    ],
    "template-curly-spacing": Array [
      "off",
    ],
    "template-tag-spacing": Array [
      "off",
    ],
    "unicode-bom": Array [
      "off",
    ],
    "use-isnan": Array [
      "error",
    ],
    "valid-typeof": Array [
      "error",
    ],
    "wrap-iife": Array [
      "off",
    ],
    "wrap-regex": Array [
      "off",
    ],
    "yield-star-spacing": Array [
      "off",
      "before",
    ],
    "yoda": Array [
      "error",
    ],
  },
  "settings": Object {},
}
`;
